src/parser/postgresparser.cpp

```diff
@@ -967,11 +967,18 @@ parser::SQLStatement *PostgresParser::CreateTransform(CreateStmt *root) {
       new CreateStatement(CreateStatement::CreateType::kTable);
   RangeVar *relation = root->relation;
   result->table_info_.reset(new parser::TableInfo());
+  if (relation->relpersistence == 't') {
+    result->is_temp_table = true;
+  }
 
   if (relation->relname) {
     result->table_info_->table_name = relation->relname;
   }
   if (relation->schemaname) {
+    if(result->is_temp_table) {
```

Comment:
Need more comments here. What does relpersistence indicate? Does it mean that the temporary table already exist? Maybe need a more expressive exception.